crates/payload/basic/src/lib.rs

```diff
@@ -587,6 +640,72 @@ fn build_payload<Pool, Client>(
+        #[cfg(feature = "optimism")]
+        {
+            for deposit_tx in attributes.transactions {
+                // Check if the job was cancelled, if so we can exit early.
+                if cancel.is_cancelled() {
+                    return Ok(BuildOutcome::Cancelled)
+                }
+
+                // Convert the deposit transaction to a [TransactionSignedEcRecovered]. This is
+                // purely for the purposes of utilizing the [tx_env_with_recovered] function,
+                // deposit transactions do not have signatures.
+                let deposit_tx = deposit_tx
+                    .clone()
+                    .try_into_ecrecovered()
+                    .map_err(|_| PayloadBuilderError::DepositTransactionRecoverFailed)?;
+
+                // Configure the environment for the block.
+                let env = Env {
+                    cfg: initialized_cfg.clone(),
+                    block: initialized_block_env.clone(),
+                    tx: tx_env_with_recovered(&deposit_tx),
+                };
+
+                let mut evm = revm::EVM::with_env(env);
+                evm.database(&mut db);
+
+                let ResultAndState { result, state } = match evm.transact() {
+                    Ok(res) => res,
+                    Err(err) => {
+                        match err {
+                            EVMError::Transaction(err) => {
+                                // if the deposit transaction is invalid, we can skip it
+                                trace!(?err, ?deposit_tx, "skipping invalid deposit transaction");
+                                continue
+                            }
+                            err => {
+                                // this is an error that we should treat as fatal for this attempt
+                                return Err(PayloadBuilderError::EvmExecutionError(err))
+                            }
+                        }
+                    }
+                };
+
+                // commit changes
+                commit_state_changes(&mut db, &mut post_state, block_number, state, true);
+
+                // Push transaction changeset and calculate header bloom filter for receipt.
+                post_state.add_receipt(
+                    block_number,
+                    Receipt {
+                        tx_type: deposit_tx.tx_type(),
+                        success: result.is_success(),
+                        cumulative_gas_used,
+                        logs: result.logs().into_iter().map(into_reth_log).collect(),
+                        #[cfg(feature = "optimism")]
+                        deposit_nonce: None,
+                    },
+                );
+
+                // append transaction to the list of executed transactions
+                executed_txs.push(deposit_tx.into_signed());
+            }
+        }
+
         while let Some(pool_tx) = best_txs.next() {
```

Comment:
This may include the deposit transactions right?

Reply:
idts, deposit transactions should never be in the tx pool